feature/experimental/system/gnmi/benchmarking/ate_tests/drained_configuration_convergence_time/drained_configuration_convergence_time_bgp_test.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/telemetry"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+const (
+	asPathRepeatValue      = 3
+	aclStatement2          = "20"
+	aclStatement3          = "30"
+	setAspathPrependPolicy = "SET-ASPATH-PREPEND"
+	setMedPolicy           = "SET-MED"
+	bgpMed                 = 25
+)
+
+// configureSetMED function us used to configure routing policy to set MED on DUT
+// TODO : SetMED is not supported: https://github.com/openconfig/featureprofiles/issues/759
+func configureSetMED(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	// clear existing policies on dut
+	dut.Config().RoutingPolicy().Replace(t, nil)
+	d := &telemetry.Device{}
+	rp := d.GetOrCreateRoutingPolicy()
+	pdef5 := rp.GetOrCreatePolicyDefinition(setMedPolicy)
+	actions5 := pdef5.GetOrCreateStatement(aclStatement3).GetOrCreateActions()
+	// TODO : SetMED is not supported: https://github.com/openconfig/featureprofiles/issues/759
+	//setMedBGP := actions5.GetOrCreateBgpActions()
+	//setMedBGP.SetMed = ygot.Uint32(bgpMed)
+	actions5.PolicyResult = telemetry.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE
+	dut.Config().RoutingPolicy().Replace(t, rp)
+}
+
+// configureSetASPath function is used to configure route policy set-as-path prepend on DUT
+func configureSetASPath(t *testing.T, dut *ondatra.DUTDevice) {
+	//clear existing policies on dut
+	dut.Config().RoutingPolicy().Replace(t, nil)
+	d := &telemetry.Device{}
+	rp := d.GetOrCreateRoutingPolicy()
+	pdef5 := rp.GetOrCreatePolicyDefinition(setAspathPrependPolicy)
+	actions5 := pdef5.GetOrCreateStatement(aclStatement2).GetOrCreateActions()
+	aspend := actions5.GetOrCreateBgpActions().GetOrCreateSetAsPathPrepend()
+	aspend.Asn = ygot.Uint32(setup.DutAS)
+	aspend.RepeatN = ygot.Uint8(asPathRepeatValue)
+	actions5.PolicyResult = telemetry.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE
+	dut.Config().RoutingPolicy().Replace(t, rp)
+}
+
+// TODO : verifyBGPAsPath is to Validate AS Path attribute using bgp rib telemetry on ATE
+// https://github.com/openconfig/ondatra/issues/45
+func verifyBGPAsPath(t *testing.T, dut *ondatra.DUTDevice) {
+	ate := ondatra.ATE(t, "ate")
+	dutPolicyConfPath := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).
+		Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp().
+		PeerGroup(setup.PeerGrpName).ApplyPolicy().ExportPolicy()
+
+	// Build wantArray to compare the diff
+	/*var wantArray []uint32
+	for i := 0; i < setup.RouteCount; i++ {
+		wantArray = append(wantArray, setup.DutAS, setup.DutAS, setup.DutAS, setup.DutAS, setup.AteAS)
+	}*/
+
+	//Start the timer.
+	start := time.Now()
+	dutPolicyConfPath.Replace(t, []string{setAspathPrependPolicy})
+	t.Run("BGP-MED-Verification", func(t *testing.T) {
+		// at := ate.Telemetry()
+		for _, ap := range ate.Ports() {
```

Comment:
Add another loop and exit only when ATE learnt all the routes? The existing code will fail if ATE doesn't learn all the routes after the change right away.

Reply:
updated, marking as resolved.